src/Umbraco.Core/Configuration/Models/InstallDefaultDataSettings.cs

```diff
@@ -0,0 +1,55 @@
+// Copyright (c) Umbraco.
+// See LICENSE for more details.
+
+using System.Collections.Generic;
+
+namespace Umbraco.Cms.Core.Configuration.Models
+{
+    public enum InstallDefaultDataOption
+    {
+        None,
+        InstallOnly,
+        InstallAllExcept,
+        All
+    }
+
+    /// <summary>
+    /// Typed configuration options for installation of default data.
+    /// </summary>
+    public class InstallDefaultDataSettings
+    {
+        /// <summary>
+        /// Gets or sets a value indicating whether to create default language data on installation.
+        /// </summary>
+        public InstallDefaultDataOption InstallData { get; set; } = InstallDefaultDataOption.All;
+
+        /// <summary>
+        /// Gets or sets a value indicating which default languages should be created when <see cref="CreateDefaultLanguages"/> is
+        /// set to <see cref="InstallDefaultDataOption.InstallOnly"/> or <see cref="InstallDefaultDataOption.InstallAllExcept"/>.
+        /// </summary>
+        /// <remarks>
+        ///     <para>
+        ///         For languages, the values provided should be the ISO codes for the languages to be included or excluded, e.g. "en-US".
+        ///         If removing the single default language, ensure that a different one is created via some other means (such
+        ///         as a restore from Umbraco Deploy schema data).
+        ///     </para>
+        ///     <para>
+        ///         For data types, the values provided should be the Guid values used by Umbraco for the data type, listed at:
+        ///         https://github.com/umbraco/Umbraco-CMS/blob/v9/dev/src/Umbraco.Core/Constants-DataTypes.cs.
+        ///         Some data types - such as the string label - cannot be excluded from install as they are required for core Umbraco
+        ///         functionality.
+        ///         Otherwise take care not to remove data types required for default Umbraco media and member types, unless you also
+        ///         choose to exclude them.
+        ///     </para>
+        ///     <para>
+        ///         For media types, the values provided should be the Guid values used by Umbraco for the media type, listed at:
+        ///         https://github.com/umbraco/Umbraco-CMS/blob/v9/dev/src/Umbraco.Infrastructure/Migrations/Install/DatabaseDataCreator.cs.
+        ///     </para>
+        ///     <para>
+        ///         For member types, the values provided should be the Guid values used by Umbraco for the media type, listed at:
+        ///         https://github.com/umbraco/Umbraco-CMS/blob/v9/dev/src/Umbraco.Infrastructure/Migrations/Install/DatabaseDataCreator.cs.
+        ///     </para>
+        /// </remarks>
+        public List<string> SelectedValues { get; set; } = new List<string>();
```

Comment:
I'd rename this to `Values`, make it an `IList` instead and remove the setter, so you can't (accidentally) set this to `null`:

```suggestion
        public IList<string> Values { get; } = new List<string>();
```

The `InstallDefaultDataOption` could then be updated to reference this name:
```c#
// Do not install any default data
None,
// Only install the default data specified in the values
Values,
// Install all default data, except the data specified in the values
ExceptValues,
// Install all default data
All
```

Reply:
All the other configuration models have public setters, so at least for consistency and perhaps for necessity, think it may be better to leave this.  Otherwise agree with your suggestions and will update.